src/leaseCreateCharge/requests.ts

```diff
@@ -7,10 +7,15 @@ export const fetchAttributes = (): Generator<any, any, any> => {
     method: 'OPTIONS'
   }));
 };
-export const fetchReceivableTypes = (): Generator<any, any, any> => {
+export const fetchReceivableTypes = (nextUrl: string): Generator<any, any, any> => {
   const state = store.getState();
   const lease = getCurrentLease(state);
   const serviceUnit = lease.service_unit;
+  if (nextUrl) {
```

Comment:
Is it possible to implement the complete logic in this function for paginating all the results?

Reply:
Probably. Requires some restructuring of function calls and return values. We would need to restructure the `call` and `callApi` invocations and return values in this function and the saga, because we can't otherwise destructure the response properly here. Using these higher order functions as responses can be a little confusing.

Reply:
It was implemented in the saga because that's the place where we already destructure the JSON data and can easily pick out the next URL.

Reply:
That does make sense. How about making a version of `callApi` that paginates all the pages? So it could be reused in place of callApi in any of these requests